services/cd-service/pkg/repository/transformer.go

```diff
@@ -2215,13 +2237,21 @@ func (c *CreateEnvironmentApplicationLock) Transform(
 	if err != nil {
 		return "", err
 	}
+	if c.CiLink != "" && state.DBHandler.ShouldUseOtherTables() && !isValidLink(c.CiLink, c.AllowedDomains) {
```

Comment:
what if it's an invalid link but `ShouldUseOtherTables` is false?
We should also handle that case, e.g. by returning an error that says this feature is only available with the db.